flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/sst/SstStats.java

```diff
@@ -0,0 +1,162 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.lsm.sst;
+
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.ArrayType;
+import org.apache.flink.table.types.logical.BigIntType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.util.Preconditions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/** Statistics for each column in a SST file. */
+public class SstStats {
+
+    private final RowData minValues;
+    private final RowData maxValues;
+    private final long[] nullCounts;
+
+    private final RowData.FieldGetter[] fieldGetters;
+
+    SstStats(RowData minValues, RowData maxValues, long[] nullCounts, RowType rowType) {
+        this(minValues, maxValues, nullCounts, RowData.createFieldGetters(rowType));
+    }
+
+    SstStats(
+            RowData minValues,
+            RowData maxValues,
+            long[] nullCounts,
+            RowData.FieldGetter[] fieldGetters) {
+        this.minValues = minValues;
+        this.maxValues = maxValues;
+        this.nullCounts = nullCounts;
+
+        this.fieldGetters = fieldGetters;
+    }
+
+    public Object minValue(int i) {
+        return fieldGetters[i].getFieldOrNull(minValues);
+    }
+
+    public Object maxValue(int i) {
+        return fieldGetters[i].getFieldOrNull(maxValues);
+    }
+
+    public long nullCount(int i) {
+        return nullCounts[i];
+    }
+
+    RowData minValues() {
+        return minValues;
+    }
+
+    RowData maxValues() {
+        return maxValues;
+    }
+
+    long[] nullCounts() {
+        return nullCounts;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof SstStats)) {
+            return false;
+        }
+        SstStats that = (SstStats) o;
+        return Objects.equals(minValues, that.minValues)
+                && Objects.equals(maxValues, that.maxValues)
+                && Arrays.equals(nullCounts, that.nullCounts);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(minValues, maxValues, nullCounts);
+    }
+
+    public static RowType schema(RowType rowType) {
+        List<RowType.RowField> fields = new ArrayList<>();
+        fields.add(new RowType.RowField("_MIN_VALUES", rowType));
+        fields.add(new RowType.RowField("_MAX_VALUES", rowType));
+        fields.add(
+                new RowType.RowField("_NULL_COUNTS", new ArrayType(false, new BigIntType(false))));
+        return new RowType(fields);
+    }
+
+    /** For collecting and calculating {@link SstStats} from a series of rows. */
+    public static class Collector {
```

Comment:
A separate class outside this class.
Sometimes we don't need this collector. (The stats can be produced from orc/parquet writer.)

Reply:
I mean we can put this class outside `SstStats`. We can also have a collector for partitions too.